src/hotspot/share/opto/loopnode.cpp

```diff
@@ -1106,7 +1107,8 @@ int PhaseIdealLoop::extract_long_range_checks(const IdealLoopTree* loop, jlong s
         jlong scale = 0;
         if (loop->is_range_check_if(if_proj, this, T_LONG, phi, range, offset, scale) &&
             loop->is_invariant(range) && loop->is_invariant(offset) &&
-            original_iters_limit / ABS(scale * stride_con) >= min_iters) {
+            original_iters_limit / ABS(scale) >= min_iters * ABS(stride_con)) {
```

Comment:
I assume there is check somewhere that `stride_con` is not `MIN_INT`.

Reply:
In my opinion ABS() should assert that it has legal input (not MIN_INT) and output (non-negative value) in debug builds.

Reply:
Thanks for reviewing this.
We can't get to this code if `stride_con` is `MIN_INT` because some other condition (that doesn't explicitly check that `stride_con` is not `MIN_INT`) causes a bail out from the transformation. I added an explicit bail out in that case in a new commit anyway to make the code more robust.

Reply:
> In my opinion ABS() should assert that it has legal input (not MIN_INT) and output (non-negative value) in debug builds.

I agree and filed [JDK-8328934](https://bugs.openjdk.org/browse/JDK-8328934) for that.

Reply:
Unfortunately, there is still an overflow here when `scale` is min_jlong.

Reply:
Right but isn't it harmless in this particular case?

Reply:
No, if it's undefined behavior, we can't be sure what result the C++ compiler will give.  And if we test with -ftrapv it will crash.

Reply:
Ok. I filed https://bugs.openjdk.org/browse/JDK-8329163 to track it.